lib/puppet/application/face_base.rb

```diff
@@ -240,13 +240,13 @@ def main
     # --daniel 2011-04-27
     if (arity = @action.positional_arg_count) > 0
       unless (count = arguments.length) == arity then
-        s = arity == 2 ? '' : 's'
-        raise ArgumentError, "puppet #{@face.name} #{@action.name} takes #{arity-1} argument#{s}, but you gave #{count-1}"
+        s = arity == 2 ? '' : _('s')
+        raise ArgumentError, _("puppet %{face} %{action} takes %{arg_count} argument%{s}, but you gave %{given_count}") % { face: @face.name, action: @action.name, arg_count: arity-1, s: s, given_count: count-1 }
```

Comment:
These two lines seem like they could be revised to handle the plural form externalization